include/sampling/sampling.hpp

```diff
@@ -21,45 +21,37 @@
 
 #ifndef SAMPLE_ONLY_H
 #define SAMPLE_ONLY_H
-
 template <typename WalkTypePolicy,
           typename PointList,
           typename Polytope,
           typename RandomNumberGenerator,
-          typename Point
-        >
+          typename Point>
 void uniform_sampling(PointList &randPoints,
-                   Polytope &P,
-                   RandomNumberGenerator &rng,
-                   const unsigned int &walk_len,
-                   const unsigned int &rnum,
-                   const Point &starting_point,
-                   unsigned int const& nburns)
-{
+                      Polytope &P,
+                      RandomNumberGenerator &rng,
+                      const unsigned int &walk_len,
+                      const unsigned int &rnum,
+                      Point &starting_point,  
+                      unsigned int const& nburns) {
 
-    typedef typename WalkTypePolicy::template Walk
-            <
-                    Polytope,
-                    RandomNumberGenerator
-            > walk;
+    typename WalkTypePolicy::template Walk<Polytope, RandomNumberGenerator> walk(P, starting_point, rng);
 
-    //RandomNumberGenerator rng(P.dimension());
-    PushBackWalkPolicy push_back_policy;
+    Point p = starting_point; 
 
-    Point p = starting_point;
-
-    typedef RandomPointGenerator <walk> RandomPointGenerator;
     if (nburns > 0) {
-        RandomPointGenerator::apply(P, p, nburns, walk_len, randPoints,
-                                    push_back_policy, rng);
-        randPoints.clear();
+        for (unsigned int i = 0; i < nburns; ++i) {
+            walk.apply(P, p, walk_len, rng);  
+        }
     }
-    RandomPointGenerator::apply(P, p, rnum, walk_len, randPoints,
-                                push_back_policy, rng);
-
 
+    for (unsigned int i = 0; i < rnum; ++i) {
+        walk.apply(P, p, walk_len, rng); 
+        randPoints.push_back(p);  
```

Comment:
Could you please reserve memory equal to rnum points before the push_backs? (i.e., randPoints.reserve(rnum);)

Reply:
I am not able to do that directly using randPoints.reserve(rnum); as randPoints is an instance of an std::list and not std::vector. Thus it does not support this type of command. Am I missing something?

Reply:
I thought about finding a workaround for reserving space from the first time, however as I figured out randPoints was a list, I decided that it is not the case anymore, as I can't get continuous storage anyway.

Reply:
`randPoints` is a templated (i.e. it can be either a std::vector or a std::list or any container that supports `push_back`), thus you can create a specialized version of this function that reserves memory only for the case of a vector but I propose not to do anything in this PR. We should think if we need to want to keep it as it is or restrict `randPoints` to std::list or std::vector.
@TolisChal do we need random access in samples? If not should we restrict to std::list here?